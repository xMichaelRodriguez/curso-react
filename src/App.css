@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;500;600;800&display=swap");

html,
body .body {
  background-color: #282828;
  color: #fafaff;
}
* {
  font-family: "Poppins", sans-serif;
}

.shadow-p {
  -webkit-box-shadow: 2px 2px 39px -3px rgba(255, 255, 255, 1);
  -moz-box-shadow: 2px 2px 39px -3px rgba(255, 255, 255, 1);
  box-shadow: 2px 2px 6px -4px rgba(255, 255, 255, 1);
}
