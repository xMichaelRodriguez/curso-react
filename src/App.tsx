import React, { useRef, useState } from "react";
import { BsCheckBox, BsFillTrashFill, BsXCircle } from "react-icons/bs";
import "./bootstrap.min.css";

interface inTask {
  name: String;
  done: Boolean;
}

export const App = (): JSX.Element => {
  const taskRef = useRef<HTMLInputElement>(null);
  const [newTask, setNewTask] = useState<string>("");
  const [tasks, setTasks] = useState<inTask[]>([]);

  const handleSubmit = (e: any): void => {
    e.preventDefault();
    if (newTask.trim().length > 6) {
      addTask(newTask);
      setNewTask("");
      taskRef.current?.focus();
    }
  };

  const addTask = (name: string): void => {
    const newTasks: inTask[] = [...tasks, { name, done: false }];
    setTasks(newTasks);
  };
  const handleToggleTask = (i: number): void => {
    const newTasks: inTask[] = [...tasks];
    newTasks[i].done = !newTasks[i].done;

    setTasks(newTasks);
  };

  const handleRemoveTask = (i: number): void => {
    const removeTast: inTask[] = [...tasks];
    removeTast.splice(i, 1);
    setTasks(removeTast);
  };
  return (
    <div className="body">
      <div className="container p-4">
        <div className="row px-4">
          <div className="col-md-6">
            <form onSubmit={handleSubmit}>
              <div className="form-group mb-3">
                <label>New Task</label>
                <input
                  className="form-control text-light"
                  type="text"
                  value={newTask}
                  name="task"
                  autoFocus
                  ref={taskRef}
                  onChange={(e) => setNewTask(e.target.value)}
                />
              </div>
              <button type="submit" className="btn btn-primary">
                <span>Save Task</span>
              </button>
            </form>
          </div>
          <div className="col-md-6 ">
            <ul className="list-group p-4">
              {tasks.map((task: inTask, i: number) => (
                <li
                  className="list-group-item mb-3   bg-dark text-light shadow-p "
                  key={i}
                >
                  <div className=" row  d-flex justify-content-between align-items-end">
                    <div className="col-md-6 text-break">
                      <h4
                        style={{ textDecoration: task.done && "line-through" }}
                      >
                        {task.name}
                      </h4>
                    </div>
                    <div className="col-md-6 d-flex flex-row-reverse justify-content-between">
                      <button
                        className="btn btn-secondary btn-lg"
                        onClick={() => {
                          handleToggleTask(i);
                        }}
                      >
                        {task.done ? <BsCheckBox /> : <BsXCircle />}
                      </button>
                      <button
                        className="btn btn-danger "
                        onClick={() => {
                          handleRemoveTask(i);
                        }}
                      >
                        <BsFillTrashFill />
                      </button>
                    </div>
                  </div>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};
