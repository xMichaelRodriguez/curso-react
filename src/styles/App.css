@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;500;800&display=swap");

* {
  font-family: "Poppins", sans-serif;
}

h1 {
  font-weight: 800;
}
h3 {
  font-weight: 600;
}
.header-text {
  margin-top: 60px;
}

.profile-section {
  background-color: #fafafa;
}

.container {
  padding-top: 20px;
}

.cover {
  z-index: -1;
}

.primary {
  background-color: #ff5300 !important;
  color: #fff;
  border-radius: 3px;
  margin-right: 10px;
  text-transform: capitalize;
}

.secondary {
  background-color: none !important;
  background: none !important;
  border: solid 1px;
  border-radius: 3px;
  border-color: #ff5300 !important;
  color: #ff5300;
  margin-right: 10px;
  text-transform: capitalize;
}

.secondary:hover {
  background-color: #ff5300 !important;
  color: #fafafa;
  box-shadow: none;
}
.offset {
  margin-top: -80px !important;
}

.primar-color-username {
  color: #ff5300;
  font-weight: 500;
}
.post-item-left-bar {
  position: absolute;
  left: 1px;
  height: 45px;
  width: 3px;
  background-color: #ff5300;
}

.push {
  min-height: calc(60vh - 100px);
}

.text-muted {
  color: #c2c2c2 !important;
}

.primary-text {
  color: #ff5300;
}

.wid {
  max-height: 600px;
}

.breadcrumb {
  background-color: transparent;
}

.activee {
  border-bottom: 3px solid #ff5300;
  border-radius: 2px;
}

.text__color {
  color: #ff5300;
}

.auth {
  position: absolute;
  display: flex;
  justify-content: center;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  max-width: 350px;
  padding: 20px;
 
}
