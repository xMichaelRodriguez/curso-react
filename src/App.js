import { AppRouter } from "./router/AppRouter";
import { Provider } from "react-redux";
import { store } from "./store/store";
import "boxicons";
import "./App.css";
function App() {
  return (
    <Provider store={store}>
      <AppRouter />;
    </Provider>
  );
}

export default App;
