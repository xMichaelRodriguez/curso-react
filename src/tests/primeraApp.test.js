import React from "react";
import { shallow } from "enzyme";
import App from "../primeraApp";

describe("Pruebas en app", () => {
  /*   test("Debe de mostrar el mensaje hola", () => {
    const saludo = "Hola Soy Tu papi";
    const {getByText} = render(<App saludo={saludo} />);

    expect(getByText(saludo)).toBeInTheDocument()
  }); */

  test("Debe de mostrar el componente <App/>", () => {
    const saludo = "Hola soy tu papi";
    const wrapper = shallow(<App saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el subtitulo pai", () => {
    const saludo = "Hola soy tu papi";
    const subtitulo = "Hola soy tu subtitulo";
    const wrapper = shallow(<App saludo={saludo} subtitulo={subtitulo} />);
    const parrafo = wrapper.find("p").text();
    /* console.log(parrafo); */
    expect(parrafo).toBe(subtitulo);
  });
});
