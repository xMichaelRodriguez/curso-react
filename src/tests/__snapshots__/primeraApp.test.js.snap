// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en app Debe de mostrar el componente <App/> 1`] = `
<Fragment>
  <h1>
    Hola soy tu papi
  </h1>
  <p>
    soy subtitulo papi
  </p>
</Fragment>
`;
